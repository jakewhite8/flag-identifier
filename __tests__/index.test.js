/* eslint-env jest */

import { shallow } from 'enzyme'
import React from 'react'
import renderer from 'react-test-renderer'

import Home from '../pages/index.js'

describe('With Enzyme', () => {
  it('App shows "Flag Identifier"', () => {
    const app = shallow(<Home />).dive()

    expect(app.find('h1').text()).toEqual('Flag Identifier')
  })
})

describe('With Snapshot Testing', () => {
  it('App shows "Flag Identifier!"', () => {
    const component = renderer.create(<Home />)
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
})
