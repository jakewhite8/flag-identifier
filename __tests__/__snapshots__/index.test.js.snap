// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`With Snapshot Testing App shows "Flag Identifier!" 1`] = `
<div>
  <h1>
    Flag Identifier
  </h1>
  <div>
    <h3>
      Color:
    </h3>
    <div
      className="colorFilters"
    >
      <a
        href="/?colors=red"
        onClick={[Function]}
      >
        red
      </a>
      <a
        href="/?colors=green"
        onClick={[Function]}
      >
        green
      </a>
      <a
        href="/?colors=blue"
        onClick={[Function]}
      >
        blue
      </a>
      <a
        href="/?colors=orange"
        onClick={[Function]}
      >
        orange
      </a>
      <a
        href="/?colors=purple"
        onClick={[Function]}
      >
        purple
      </a>
      <a
        href="/?colors=yellow"
        onClick={[Function]}
      >
        yellow
      </a>
      <a
        href="/?colors=white"
        onClick={[Function]}
      >
        white
      </a>
      <a
        href="/?colors=black"
        onClick={[Function]}
      >
        black
      </a>
    </div>
  </div>
  <div>
    <img
      src="https://www.countryflags.io/AM/flat/64.png"
    />
    <img
      src="https://www.countryflags.io/AL/flat/64.png"
    />
    <img
      src="https://www.countryflags.io/BE/flat/64.png"
    />
    <img
      src="https://www.countryflags.io/BT/flat/64.png"
    />
  </div>
</div>
`;
